### PR TITLE
ss-68: Add API to update user profile data

### PR DESCRIPTION
**Pull request by Holovashcenko**
- Added user profile migration 
- Implemented ability to update user profile

image
image

### CODE REVIEW

**Comment by Holovashcenko** [comment]
image
image

File: backend/src/bundles/users/types/user-gender.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+type Gender = 'male' | 'female';
+
+export { type Gender };
```
Comment (teslmik) [inline backend/src/bundles/users/types/user-gender.ts:3]: Can it be better to make an enum?

File: backend/src/bundles/users/user.controller.ts [hunk 2]
```diff
@@ -37,6 +40,14 @@ class UserController extends Controller {
             method: 'GET',
             handler: () => this.findAll(),
         });
+
+        this.addRoute({
+            path: `${UsersApiPath.ROOT}`,
+            method: 'PUT',
+            handler: (options) => {
+                return this.update(options as UpdateRequestDto);
```
Comment (teslmik) [inline backend/src/bundles/users/user.controller.ts:48]: No backticks needed here

**Review by teslmik — APPROVED** [review]

File: backend/src/bundles/users/types/user-update-request-dto.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { type Gender } from './types.js';
+
+type UserUpdateRequestDto = {
```
Comment (nikita-remeslov) [inline backend/src/bundles/users/types/user-update-request-dto.ts:3]: for update better use Partial

File: backend/src/bundles/users/types/user-update-response-dto.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { type Gender } from './types.js';
+
+type UserUpdateResponseDto = {
```
Comment (nikita-remeslov) [inline backend/src/bundles/users/types/user-update-response-dto.ts:3]: Partial

File: backend/src/bundles/users/user.controller.ts [hunk 5]
```diff
@@ -24,6 +26,7 @@ import { UsersApiPath } from './enums/enums.js';
  *            type: string
```
Comment (nikita-remeslov) [inline backend/src/bundles/users/user.controller.ts:26]: update user entity

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

File: backend/src/bundles/users/types/user-gender.ts (hunk 1 above)
Comment (Holovashcenko) [reply backend/src/bundles/users/types/user-gender.ts]: fix in [https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-smartspend/pull/98/commits/14fae07af2bc0621f3e3db570ea9f67bec727e50](url)

File: backend/src/bundles/users/user.controller.ts (hunk 2 above)
Comment (Holovashcenko) [reply backend/src/bundles/users/user.controller.ts]: fix in [https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-smartspend/pull/98/commits/3606ef6d2bc9fa2faa9acfe9d665af47df196d7a](url)

File: backend/src/bundles/users/types/user-update-request-dto.ts (hunk 3 above)
Comment (Holovashcenko) [reply backend/src/bundles/users/types/user-update-request-dto.ts]: fix in [https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-smartspend/pull/98/commits/bcf52b7e3b0dca9d81bd78c4ca1be95523bc8c28](url)

File: backend/src/bundles/users/types/user-update-response-dto.ts (hunk 4 above)
Comment (Holovashcenko) [reply backend/src/bundles/users/types/user-update-response-dto.ts]: fix in [https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-smartspend/pull/98/commits/bcf52b7e3b0dca9d81bd78c4ca1be95523bc8c28](url)

File: backend/src/bundles/users/user.controller.ts (hunk 5 above)
Comment (Holovashcenko) [reply backend/src/bundles/users/user.controller.ts:26]: fix in [https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-smartspend/pull/98/commits/5b2b613ffbe6869c2f7f6186207b4c9bf8f25abe](url)

File: backend/src/bundles/users/types/user-update-request-dto.ts [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import { type Gender } from './types.js';
+
+type UserUpdateRequestDto = Partial<{
```
Comment (nikita-remeslov) [inline backend/src/bundles/users/types/user-update-request-dto.ts:3]: I not sure that you need entity with all not requered field

File: backend/src/bundles/users/user.controller.ts [hunk 7]
```diff
@@ -61,6 +72,74 @@ class UserController extends Controller {
...
+     *               $ref: '#/components/schemas/User'
+     *   components:
+     *     securitySchemes:
+     *       bearerAuth:
+     *         type: http
+     *         scheme: bearer
+     */
+
+    private async update(
+        request: UpdateRequestDto,
```
Comment (nikita-remeslov) [inline backend/src/bundles/users/user.controller.ts:134]:
```suggestion
        request: Partial<UpdateRequestDto>,
```

File: backend/src/bundles/users/types/user-update-request-dto.ts (hunk 6 above)
Comment (nikita-remeslov) [reply backend/src/bundles/users/types/user-update-request-dto.ts:3]: Create normal entity, without Partial

File: backend/src/bundles/users/types/user-update-response-dto.ts [hunk 8]
```diff
@@ -0,0 +1,14 @@
+import { type Gender } from './types.js';
+
+type UserUpdateResponseDto = Partial<{
```
Comment (nikita-remeslov) [inline backend/src/bundles/users/types/user-update-response-dto.ts:3]: Create normal entity, without Partial

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

File: backend/src/bundles/users/types/user-update-response-dto.ts (hunk 8 above)
Comment (Holovashcenko) [reply backend/src/bundles/users/types/user-update-response-dto.ts]: fix in [https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-smartspend/pull/98/commits/62a76c664ba69ef6318c1757a3e2d6e18969cec6](url)

File: backend/src/bundles/users/user.controller.ts (hunk 7 above)
Comment (Holovashcenko) [reply backend/src/bundles/users/user.controller.ts:134]: image

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**